web3/contracts.go

```diff
@@ -145,9 +162,44 @@ func (c *Contracts) LoadContracts(addresses *Addresses) error {
 	if err != nil {
 		return fmt.Errorf("failed to bind process registry: %w", err)
 	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), web3QueryTimeout)
+	defer cancel()
+
+	// check vkey used by sequencer is the same as the one used by the contract
+	vkey, err := process.GetVerifierVKeyHash(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return fmt.Errorf("failed to get verifier address: %w", err)
+	}
+	svkey := hex.EncodeToString(vkey[:])
+	svkey = strings.TrimPrefix(svkey, "0x")
+	if svkey != config.StateTransitionProvingKeyHash {
+		return fmt.Errorf("verifier vkey hash mismatch: %s != %s", vkey, config.StateTransitionProvingKeyHash)
```

Comment:
This error message is not correct. Maybe "proving key hash mismatch with the one provided by the smart contract"